langchain-ai/langchain
# Add Support for Loading Documents from Huawei OBS

Description:
This PR adds support for loading documents from Huawei OBS (Object Storage Service) in Langchain. OBS is a cloud-based object storage service provided by Huawei Cloud. With this enhancement, Langchain users can now easily access and load documents stored in Huawei OBS directly into the system.

Key Changes:
- Added a new document loader module specifically for Huawei OBS integration.
- Implemented the necessary logic to authenticate and connect to Huawei OBS using access credentials.
- Enabled the loading of individual documents from a specified bucket and object key in Huawei OBS.
- Provided the option to specify custom authentication information or obtain security tokens from Huawei Cloud ECS for easy access.

How to Test:
1. Ensure the required package "esdk-obs-python" is installed.
2. Configure the endpoint, access key, secret key, and bucket details for Huawei OBS in the Langchain settings.
3. Load documents from Huawei OBS using the updated document loader module.
4. Verify that documents are successfully retrieved and loaded into Langchain for further processing.

Please review this PR and let us know if any further improvements are needed. Your feedback is highly appreciated!

@rlancemartin, @eyurtsev
